### PR TITLE
Update to ZIO 2.0-RC3

### PR DESCRIPTION
**Pull request by DamianReeves**

### CODE REVIEW

**Comment by DamianReeves** [comment]
@adamgfraser could you take a look at this?

File: zio-cli/shared/src/main/scala/zio/cli/PrimType.scala [hunk 1]
```diff
@@ -327,7 +327,7 @@ object PrimType {
     def validate(value: Option[String], conf: CliConfig): IO[String, JYearMonth] = {
       val AcceptedFormat = "^(-?\\d+)-(\\d{2})".r
       def parse(input: String) = input match {
-        case AcceptedFormat(y, m) => ZIO(JYearMonth.of(y.toInt, m.toInt))
+        case AcceptedFormat(y, m) => ZIO.succeed(JYearMonth.of(y.toInt, m.toInt))
```
Comment (adamgfraser) [inline zio-cli/shared/src/main/scala/zio/cli/PrimType.scala:330]: Can this throw exceptions? In prior ZIO versions `ZIO.apply` was equivalent to `ZIO.attempt` and `UIO.apply` was equivalent to `UIO.succeed` so `attempt` would actually be the more faithful translation here.

**Review by adamgfraser — APPROVED** [review]
Looks good. A couple of minor comments.

File: zio-cli/shared/src/main/scala/zio/cli/PrimType.scala (hunk 1 above)
Comment (DamianReeves) [reply zio-cli/shared/src/main/scala/zio/cli/PrimType.scala]: Done

File: zio-cli/jvm/src/test/scala/zio/cli/figlet/FigFontRenderReportSpec.scala [hunk 2]
```diff
@@ -55,30 +57,32 @@ object FigFontRenderReportSpec extends ZIOSpecDefault {
...
-              w.write(s)
-              w.write("```\n")
-            case (url, Right(lines)) =>
-              w.write(s"\n* <$url>\n```\n")
-              lines.foreach { l =>
-                w.write(l)
-                w.write('\n')
-              }
-              w.write("```\n")
+      zio.ZIO.scoped[Any] {
```
[the remark this answers is not in the document]
Comment (DamianReeves) [reply zio-cli/jvm/src/test/scala/zio/cli/figlet/FigFontRenderReportSpec.scala]: Done

File: zio-cli/jvm/src/test/scala/zio/cli/figlet/FigFontRenderReportSpec.scala [hunk 3]
```diff
@@ -18,9 +18,11 @@ object FigFontRenderReportSpec extends ZIOSpecDefault {
   override def spec = suite("FigFontRenderReportSpec")(
     test("figlet.org Fonts Render Report") {
       for {
-        fontDbHtml <- ZManaged
-                        .acquireReleaseWith(attemptBlockingIO(Source.fromURL(fontDbUrl)))(source => UIO(source.close()))
-                        .use(s => attemptBlockingIO(s.getLines().mkString))
+        fontDbHtml <- ZIO.scoped {
```
[the remark this answers is not in the document]
Comment (DamianReeves) [reply zio-cli/jvm/src/test/scala/zio/cli/figlet/FigFontRenderReportSpec.scala]: Done

**Comment by bilal-fazlani** [comment]
please release this to maven 🥺

**Pull request merged**